event-reader-sink/src/main/kotlin/gov/cdc/ocio/eventreadersink/sink/AzureSink.kt

```diff
@@ -0,0 +1,128 @@
+package gov.cdc.ocio.eventreadersink.sink
+
+import gov.cdc.ocio.eventreadersink.camel.AzureRoutes
+import gov.cdc.ocio.eventreadersink.exceptions.BadServiceException
+import gov.cdc.ocio.eventreadersink.exceptions.BadStateException
+import gov.cdc.ocio.eventreadersink.exceptions.ConfigurationException
+import org.apache.camel.CamelContext
+import org.apache.camel.impl.DefaultCamelContext
+import org.apache.qpid.jms.JmsConnectionFactory
+import org.apache.camel.component.amqp.AMQPComponent
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * AzureSink handles the configuration and setup for sinking messages
+ * from an Azure Service Bus topic subscription to Azure Blob Storage.
+ */
+class AzureSink {
+
+    /**
+     * Sinks messages from an Azure Service Bus topic subscription to Blob Storage.
+     *
+     * This method initializes the Camel context, configures the AMQP component,
+     * and adds the necessary routes for processing incoming messages.
+     *
+     * @param connectionString The connection string for the Azure Service Bus.
+     * @param accountName The Azure storage account name.
+     * @param accountKey The Azure storage account key.
+     * @param containerName The name of the Blob Storage container.
+     * @param serviceBusNamespace The namespace for the Azure Service Bus.
+     * @param sharedAccessKeyName The shared access key name for the Service Bus.
+     * @param sharedAccessKey The shared access key for the Service Bus.
+     * @param topicName The name of the Azure Service Bus topic.
+     * @param subscriptionName The name of the subscription to the topic.
+     * @throws BadServiceException
+     */
+    @Throws(BadServiceException:: class)
+    fun sinkAsbTopicSubscriptionToBlob(
+        connectionString: String,
+        accountName: String,
+        accountKey: String,
+        containerName: String,
+        serviceBusNamespace: String,
+        sharedAccessKeyName: String,
+        sharedAccessKey: String,
+        topicName: String,
+        subscriptionName: String
+    ) {
+
+        try {
+            val camelContext: CamelContext = DefaultCamelContext()
+
+            configureAmqpComponent(
+                connectionString,
+                camelContext,
+                serviceBusNamespace,
+                sharedAccessKeyName,
+                sharedAccessKey
+            )
+            camelContext.addRoutes(AzureRoutes(topicName, subscriptionName, accountName, accountKey, containerName))
+            startCamelContext(camelContext)
+        } catch (e: Exception) {
+            logger.error("Failed to sink messages from Azure Service Bus to Blob Storage: ${e.message}", e)
+            throw BadServiceException("Error initializing AzureSink: ${e.message}")
```

Comment:
Same issue - catch BaseServiceException and have another catch for Exception

Reply:
Updated.